upup/pkg/fi/nodeup/command.go

```diff
@@ -380,6 +385,36 @@ func evaluateHostnameOverride(hostnameOverride string) (string, error) {
 	return hostnameOverride, nil
 }
 
+func evaluateBindAddress(bindAddress string) (string, error) {
+	if bindAddress == "" {
+		return "", nil
+	}
+	if bindAddress == "@aws" {
+		vBytes, err := vfs.Context.ReadFile("metadata://aws/meta-data/local-ipv4")
+		if err != nil {
+			return "", fmt.Errorf("error reading local IP from AWS metadata: %v", err)
+		}
+
+		// The local-ipv4 gets it's IP from the AWS.
+		// For now just choose the first one.
+		ips := strings.Fields(string(vBytes))
+		if len(ips) == 0 {
+			glog.Warningf("Local IP from AWS metadata service was empty")
+			return "", nil
+		} else {
+			ip := ips[0]
```

Comment:
Suggestion for next time - it could be helpful to log a warning when `len(ips) > 0`.  That sort of log message makes debugging easier because there's a log message for the "first IP" assumption, rather than having to remember that nuance of the code.